Source/Mail/MailBase.pas

```diff
@@ -0,0 +1,338 @@
+unit MailBase;
+
+interface
+
+uses
+  System.SysUtils,
+  System.Classes,
+  MailInterface;
+
+type
+  TMailBase = class abstract(TInterfacedObject, IMail)
+    private
+      FHost: string;
+      FPort: Integer;
+      FUsername: string;
+      FPassword: string;
+      FSSL: Boolean;
+      FTLS: Boolean;
+      FAuthentication: Boolean;
+      FFromName: string;
+      FFromAddress: string;
+      FReplyToName: string;
+      FReplyToAddress: string;
+      FToRecipients: TStringList;
+      FCcRecipients: TStringList;
+      FBccRecipients: TStringList;
+      FConfirmation: Boolean;
+      FAttachments: TStringList;
+      FSubject: string;
+      FMessages: TStringList;
+      FHTML: Boolean;
+    protected
+      function GetHost: string;
+      function GetPort: Integer;
+      function GetUsername: string;
+      function GetPassword: string;
+      function IsWithSSL: Boolean;
+      function IsWithTLS: Boolean;
+      function IsWithAuthentication: Boolean;
```

Comment:
No caso das variáveis **IsWithConfirmation**, **IsWithAuthentication**, **IsWithSSL** e **IsWithTLS** acho que podemos tirar o Is, porque fica redundante.